pandas/core/window.py

```diff
@@ -671,14 +679,24 @@ def _apply_window(self, mean=True, **kwargs):
         window = self._prep_window(**kwargs)
         center = self.center
 
-        blocks, obj, index = self._create_blocks()
+        blocks_dict, obj, index = self._create_blocks()
+        dtypes = blocks_dict.keys()
+        blocks = blocks_dict.values()
+
         results = []
-        for b in blocks:
+        exclude = []
+        for dtype in list(dtypes):
```

Comment:
best just to
```
for b in bocks_dict.values():
    .....
```

then don't need anything else

Reply:
Since my solution to unordered dict issue requires deleting nuisance blocks (block of columns with same type) I needed a shallow copy of keys to remove them iteratively